backend/spiritual_avatar_generation_engine.py

```
@@ -246,16 +271,24 @@ async def generate_avatar_preview_lightweight(
 _avatar_engine_instance = None
 
 # REFRESH.MD: Corrected the function signature which was mistakenly altered
-def get_avatar_engine() -> "SpiritualAvatarGenerationEngine":
+# REFRESH.MD: The dependency injector now accepts the storage_service to pass to the engine.
+def get_avatar_engine(
+    storage_service: SupabaseStorageService = Depends(get_storage_service)
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)